File: wheystore/wheystore.Web/Modules/SanPhamDB/DonHang/DonHangForm.cs
namespace wheystore.SanPhamDB.Forms;

[FormScript("SanPhamDB.DonHang")]
[BasedOnRow(typeof(DonHangRow), CheckNames = true)]
public class DonHangForm
{
    public int MaKhachHang { get; set; }
    public string HoTenNguoiNhan { get; set; }
    public string DiaChiNguoiNhan { get; set; }
    public string SoDienThoaiNguoiNhan { get; set; }
    public string EmailNguoiNhan { get; set; }
    public string TinhThanhNguoiNhan { get; set; }
    public string QuanHuyenNguoiNhan { get; set; }
    public string PhuongXaNguoiNhan { get; set; }
    public string MaBuuChinhNguoiNhan { get; set; }
    public int ShipperId { get; set; }
    public DateTime NgayDatHang { get; set; }
    public DateTime NgayGiaoHangDuKhien { get; set; }
    public DateTime NgayGiaoHangHoanThanh { get; set; }
    public decimal TongTienHang { get; set; }
    public decimal PhiVanChuyen { get; set; }
    public decimal TongThanhToan { get; set; }
    public decimal GiamGiaDonHang { get; set; }
    public string PhuongThucThanhToan { get; set; }
    public bool DaThanhToan { get; set; }
    public short TrangThaiDonHang { get; set; }
    public string GhiChu { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/DonHangForm.ts
import { BooleanEditor, DateEditor, DecimalEditor, initFormType, LookupEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface DonHangForm {
    MaKhachHang: LookupEditor;
    HoTenNguoiNhan: StringEditor;
    DiaChiNguoiNhan: StringEditor;
    SoDienThoaiNguoiNhan: StringEditor;
    EmailNguoiNhan: StringEditor;
    ShipperId: LookupEditor;
    NgayDatHang: DateEditor;
    NgayGiaoHangDuKhien: DateEditor;
    NgayGiaoHangHoanThanh: DateEditor;
    TongTienHang: DecimalEditor;
    PhiVanChuyen: DecimalEditor;
    TongThanhToan: DecimalEditor;
    GiamGiaDonHang: DecimalEditor;
    PhuongThucThanhToan: StringEditor;
    DaThanhToan: BooleanEditor;
    GhiChu: StringEditor;
    CreatedAt: DateEditor;
    UpdatedAt: DateEditor;
}

export class DonHangForm extends PrefixedContext {
    static readonly formKey = 'SanPhamDB.DonHang';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DonHangForm.init) {
            DonHangForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = DateEditor;
            var w3 = DecimalEditor;
            var w4 = BooleanEditor;

            initFormType(DonHangForm, [
                'MaKhachHang', w0,
                'HoTenNguoiNhan', w1,
                'DiaChiNguoiNhan', w1,
                'SoDienThoaiNguoiNhan', w1,
                'EmailNguoiNhan', w1,
                'ShipperId', w0,
                'NgayDatHang', w2,
                'NgayGiaoHangDuKhien', w2,
                'NgayGiaoHangHoanThanh', w2,
                'TongTienHang', w3,
                'PhiVanChuyen', w3,
                'TongThanhToan', w3,
                'GiamGiaDonHang', w3,
                'PhuongThucThanhToan', w1,
                'DaThanhToan', w4,
                'GhiChu', w1,
                'CreatedAt', w2,
                'UpdatedAt', w2
            ]);
        }
    }
}